compiler/lib/src/main/scala/util/Error.scala

```diff
@@ -206,6 +209,8 @@ sealed trait Error {
         System.err.println(loc2)
       case SemanticError.PassiveAsync(loc) =>
         Error.print (Some(loc)) ("passive component may not have async input")
+      case SemanticError.PassiveStateMachine(loc) =>
```

Comment:
I think the error should be PassiveStateMachineInstance. I think a passive component can have a state machine definition, right?

Reply:
I think we should disallow state machine definitions in a passive component because it's confusing as we don't allow state machine instances in a passive component.

Reply:
I agree. You can update the spec, update the checking, and add tests.